src/test/java/dev/stalla/model/ItunesCategoryFactoryTest.java

```diff
@@ -0,0 +1,128 @@
+package dev.stalla.model;
+
+import dev.stalla.model.itunes.ItunesCategory;
+import dev.stalla.model.itunes.NestedItunesCategory;
+import dev.stalla.model.itunes.SimpleItunesCategory;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ItunesCategoryFactoryTest {
+
+    @Test
+    @DisplayName("should get the correct iTunes category instances from the interface factory method")
+    public void testItunesCategoryFactory() {
+        assertEquals(SimpleItunesCategory.ARTS, ItunesCategory.ARTS);
+        assertEquals(NestedItunesCategory.BOOKS, ItunesCategory.BOOKS);
```

Comment:
These tests here ensure we did not mess up the delegation in the `ItunesCategory` interface companion object.